templates/contribute/pr-review.md

```diff
@@ -0,0 +1,723 @@
+      or touches on too many different topics?
+- [are there obvious improvements that can be made?](#improvements)
+    + can pieces be split into supporting lemmas or definitions (especially for long proofs)?
+    + can different / better tactics be used to improve readability (e.g., using `gcongr`
+      instead of `mul_le_mul_of_nonneg_left`)?
+      Note: code golfing is okay as long as it *doesn't sacrifice readability*, although golfing
+      trivial results is generally okay.
+    + does a different proof structure greatly simplify the argument?
+    + are the definitions introduced the best way to formalize the concept (very difficult!)?
+- [does it advance or improve the library?](#library-integration)
+    + does it provide a sensible API?
+    + is it general enough to support known future needs?
+    + does it fit the design and collective vision of mathlib?
+
+- more specific considerations
+    + are lemmas tagged `@[simp]`, `@[ext]`, et cetera where they should be? or shouldn't be?
+    + do any newly declared instances create diamonds? non-defeq or non-propeq?
+- issues specific to meta code
+
+## Examples
+
+The remainder of this guide is devoted to considering both contrived and real-world
+examples of the review process. We attempt to provide examples for each of the following
+questions above. Parties involved in the real-world examples have been consulted for
+permission for inclusion in this style guide.
+
+Not every question has a corresponding example, but we attempt to provide at least a 
+discussion about what should be considered in each case.
+
+### Style
+
+#### Code formatting
+
+```lean
+theorem mul_assoc_assoc {α : Type _} [Semigroup α] (a b c d : α)
+: a*b*c*d=a*(b*(c*d)) :=
+by
+rw [mul_assoc,
+    mul_assoc]
+```
+
+The code above violates several formatting guidelines: there are no
+spaces around binary operations, the line starts with a `:` instead of 
+ending it on the previous line, `by` should be moved to the previous line
+instead of by itself (a style linter should catch this anyway in CI),
+and the `rw` tactic is unnecessarily split across multiple lines.
+
+The author of the PR in this situation, by virtue of violating so many
+style guidelines, is probably a new contributor and is unfamiliar with,
+or doesn't recall, the style guide. An appropriate review comment would 
+be something like:
+
+````markdown
+In case you're unaware, please familiarize yourself with the mathlib
+[style guide](https://leanprover-community.github.io/contribute/style.html).
+You need spaces around `*`, `:` at the end of the line and the `rw` to 
+be on the same line.
+```suggestion
+theorem mul_assoc_assoc {α : Type _} [Semigroup α] (a b c d : α) :
+    a * b * c * d = a * (b * (c * d)) := by
+  rw [mul_assoc, mul_assoc]
+```
+````
+
+#### Naming conventions
+
+```lean
+theorem inv_is_unit_times_self_eq_1 {M : Type _} [Monoid M] {a : M} (h : IsUnit a) :
```

Comment:
A general comment; can we globally replace
```suggestion
theorem inv_is_unit_times_self_eq_1 {M : Type*} [Monoid M] {a : M} (h : IsUnit a) :
```
and perhaps even add a note about `Type _` vs `Type*`?